### ISSUE TITLE
No trigger on current elect. tariff

### ISSUE THREAD

**Issue by geedsen** [fsaris/home-assistant-zonneplan-one#123]
I had this question in the Discussions, but it does not look like anyone is reading it there. So I will just ask it here:

Possibly I do not fully understand how the automations work. But I added an automation where the trigger is based on sensor.zonneplan_current_electricity_tariff . It automates three powerplugs, that (in the future) will switch off the DC side of three inverters. So when the price is negative I switch of the panels. But to test I set the state of the powerplugs to 'on' (= DC is off) and was expecting that the next hour the tariff changes it would trigger the automation "price above 0" , but it did not trigger at all.

![image](https://github.com/fsaris/home-assistant-zonneplan-one/assets/1316595/c53de989-c8c8-4010-9f45-9017a5e5a300)
So what am I doing wrong? I can see in the states of the entity that it shows the current tariff.

I added the trigger as condition as well. And used a template for the trigger. When I test the automation (using conditions) it works. But somehow I would expect this automation to be triggered every hour and it is not.

**Comment by geedsen** [comment]
Is it because the value was already above the threshold? I will try the trigger on any state change.

**Comment by martijnrusschen** [comment]
This is a general home assistant question, try asking it here: https://gathering.tweakers.net/forum/list_messages/2165986. What have you done trying to debug this? Is the value changing in your home assistant? What if you change the threshold to 0.2? Does that trigger something?

**Comment by geedsen** [comment]
I did not think it was a general question because I have other automations working fine. Just this one not so I assumed it was something with the zonneplan sensor itself. But it indeed turns out it is indeed a misunderstanding, numeric state triggers only when the threshold is passed AND changed (from not passed to passed). In my case it was passed but before the event the threshold was passed as well. I now changed it to a trigger that firtes on every state change and made the numeric test part of the condition. That seems to work.
